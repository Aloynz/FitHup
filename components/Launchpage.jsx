import React from "react";
import { Image, StatusBar, StyleSheet, Text, View } from "react-native";
import colours from "../assets/colours/colours";
import { useFonts } from "expo-font";
import Style from "./Style";
import Button from "./Button";

export default LaunchPage = ({ navigation }) => {
  const [loaded] = useFonts({
    RobotoMedium: require("../assets/fonts/Roboto-Medium.ttf"),
    ZenDots: require("../assets/fonts/ZenDots-Regular.ttf"),
  });

  return (
    <View style={styles.container}>
      <StatusBar />
      <Image
        source={require("../assets/images/LaunchPageImage.jpg")}
        style={styles.launchImage}
      />

      <Text style={styles.name}>FitHup</Text>

      <View style={[Style.loginOrSignUpButton, { marginTop: 15 }]}>
        <Button title={"Login"} onPress={() => navigation.navigate("Login")} />
      </View>

      <View style={[Style.loginOrSignUpButton, { marginTop: 23 }]}>
        <Button
          title={"Sign Up"}
          onPress={() => navigation.navigate("Sign Up")}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colours.background,
  },

  launchImage: {
    width: 470,
    height: 631 - StatusBar.currentHeight,
    opacity: 0.3,
  },

  name: {
    color: colours.text,
    fontFamily: "ZenDots",
    fontSize: 48,
    position: "absolute",
    marginTop: 105,
    marginLeft: 100,
  },
});
