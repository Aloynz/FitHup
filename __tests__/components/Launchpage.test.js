import React from 'react';
import Launchpage from '../../components/Launchpage'
import renderer from 'react-test-renderer';
import { render, waitFor, act } from '@testing-library/react-native';

describe('Launchpage', () => {
  const launchPage = renderer.create(
    <Launchpage />
  ).toJSON;

  it('should render successfully', async () => {
    await act(async () => {
      expect(launchPage).toBeDefined();
    })
  });

  it('should render correctly', async () => {
    await act(async () => {
      expect(launchPage).toMatchSnapshot();
    })
  });

  const { getByTestId } = render(<Launchpage />);

  it('should have login button', () => {
    waitFor(() => {
      const loginButton = getByTestId('launchLoginButton');
      expect(loginButton).toBeInTheDocument();
    })
  });

  it('should have sign up button', () => {
    waitFor(() => {
      const signUpButton = getByTestId('launchSignupButton');
      expect(signUpButton).toBeInTheDocument();
    })
  });
});