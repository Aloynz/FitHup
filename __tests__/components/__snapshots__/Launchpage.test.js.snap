// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launchpage should render correctly 1`] = `[Function]`;
